infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java

```diff
@@ -67,49 +77,163 @@ public class LogReplicationConfigManager {
+    // some tables opened and confirmedNoisyStreams will be read with potential conflict info)
+    private List<Map.Entry<TableName, CorfuRecord<TableDescriptors, TableMetadata>>> registryTableEntries;
+
     /**
-     * Used for testing purpose only.
+     * Used for non-upgrade testing purpose only. Note that this constructor will keep the version table in
+     * uninitialized state, in which case LR will be constantly considered to be not upgraded.
      */
     @VisibleForTesting
     public LogReplicationConfigManager(CorfuRuntime runtime) {
+        this.rt = runtime;
         this.corfuStore = new CorfuStore(runtime);
         this.pluginConfigFilePath = EMPTY_STR;
     }
 
     public LogReplicationConfigManager(CorfuRuntime runtime, String pluginConfigFilePath) {
+        this.rt = runtime;
         this.pluginConfigFilePath = pluginConfigFilePath;
         this.corfuStore = new CorfuStore(runtime);
+        initLogReplicationVersionPlugin();
+        setupVersionTable();
+    }
 
-        initStreamNameFetcherPlugin();
-        openVersionTable();
+    private void initLogReplicationVersionPlugin() {
+        log.info("Plugin :: {}", pluginConfigFilePath);
+        LogReplicationPluginConfig config = new LogReplicationPluginConfig(pluginConfigFilePath);
+        File jar = new File(config.getStreamFetcherPluginJARPath());
+        try (URLClassLoader child = new URLClassLoader(new URL[]{jar.toURI().toURL()}, this.getClass().getClassLoader())) {
+            Class plugin = Class.forName(config.getStreamFetcherClassCanonicalName(), true, child);
+            logReplicationVersionAdapter = (ILogReplicationVersionAdapter) plugin.getDeclaredConstructor()
+                    .newInstance();
+        } catch (Exception e) {
+            log.error("Fatal error: Failed to get Log Replicatior Version Plugin", e);
+            throw new UnrecoverableCorfuError(e);
+        }
     }
 
-    private void openVersionTable() {
+    /**
+     * Read registry table's entries into memory for config synchronization.
+     */
+    public void refreshInMemoryRegistryTableEntries() {
+        try {
+            IRetry.build(IntervalRetry.class, () -> {
+                try {
+                    CorfuTable<TableName, CorfuRecord<TableDescriptors, TableMetadata>> registryTable =
+                            rt.getTableRegistry().getRegistryTable();
+                    registryTableEntries = registryTable.entryStream().collect(Collectors.toList());
+                } catch (Exception e) {
+                    log.error("Exception caught when syncing with registry table, retry needed", e);
+                    throw new RetryNeededException();
+                }
+                return null;
+            }).run();
+        } catch (InterruptedException e) {
+            log.error("Cannot sync with registry table, LR stopped", e);
+            throw new UnrecoverableCorfuInterruptedError(e);
+        }
+    }
+
+
+    /**
+     * Get streams to replicate.
```

Comment:
the streams to replicate is constructed from a cached copy of the registry table.  Please update the comment accordingly